op-e2e/artifactsfs_test.go

```diff
@@ -0,0 +1,43 @@
+package op_e2e
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+
+	"github.com/ethereum/go-ethereum/log"
+
+	"github.com/ethereum-optimism/optimism/op-chain-ops/foundry"
+	"github.com/ethereum-optimism/optimism/op-service/testlog"
+)
+
+func TestArtifacts(t *testing.T) {
+	logger := testlog.Logger(t, log.LevelWarn) // lower this log level to get verbose test dump of all artifacts
+	af := foundry.OpenArtifactsDir("../packages/contracts-bedrock/forge-artifacts")
+	artifacts, err := af.ListArtifacts()
+	require.NoError(t, err)
+	require.NotEmpty(t, artifacts)
+	for _, name := range artifacts {
+		contracts, err := af.ListContracts(name)
+		require.NoError(t, err, "failed to list %s", name)
+		require.NotEmpty(t, contracts)
+		for _, contract := range contracts {
+			artifact, err := af.ReadArtifact(name, contract)
+			if err != nil {
+				if errors.Is(err, foundry.ErrLinkingUnsupported) {
+					logger.Info("linking not supported", "name", name, "contract", contract, "err", err)
+					continue
+				}
+				require.NoError(t, err, "failed to read artifact %s / %s", name, contract)
+			}
+			logger.Info("artifact",
```

Comment:
Do we still want to log this?

Reply:
The log level at the top of the test is set to Warn, with a comment about it. It's there so you can see what it does if running locally. For automation we have the `NotEmpty` checks to ensure that it's looping over data.